telegram/ext/_utils/webhookhandler.py

```diff
@@ -43,131 +40,126 @@
 
-    def serve_forever(self, ready: Event = None) -> None:
-        with self.server_lock:
-            IOLoop().make_current()
-            self.is_running = True
-            self.logger.debug('Webhook Server started.')
-            self.loop = IOLoop.current()
-            self.http_server.listen(self.port, address=self.listen)
+    async def serve_forever(self, ready: asyncio.Event = None) -> None:
+        async with self._server_lock:
+            self._http_server.listen(self.port, address=self.listen)
 
+            self.is_running = True
             if ready is not None:
                 ready.set()
 
-            self.loop.start()
-            self.logger.debug('Webhook Server stopped.')
-            self.is_running = False
+            self._logger.debug('Webhook Server started.')
 
-    def shutdown(self) -> None:
-        with self.shutdown_lock:
+    async def shutdown(self) -> None:
+        async with self._shutdown_lock:
             if not self.is_running:
-                self.logger.warning('Webhook Server already stopped.')
+                self._logger.debug('Webhook Server is already shut down. Returning')
                 return
-            self.loop.add_callback(self.loop.stop)  # type: ignore
+            self.is_running = False
+            self._http_server.stop()
+            await self._http_server.close_all_connections()
+            self._logger.debug('Webhook Server stopped')
 
     # pylint: disable=unused-argument
     def handle_error(self, request: object, client_address: str) -> None:
         """Handle an error gracefully."""
-        self.logger.debug(
+        self._logger.debug(
             'Exception happened during processing of request from %s',
             client_address,
             exc_info=True,
         )
 
 
 class WebhookAppClass(tornado.web.Application):
-    def __init__(self, webhook_path: str, bot: 'Bot', update_queue: Queue):
+    """Application used in the Webserver"""
+
+    def __init__(self, webhook_path: str, bot: 'Bot', update_queue: asyncio.Queue):
         self.shared_objects = {"bot": bot, "update_queue": update_queue}
-        handlers = [(rf"{webhook_path}/?", WebhookHandler, self.shared_objects)]  # noqa
+        handlers = [(rf"{webhook_path}/?", TelegramHandler, self.shared_objects)]  # noqa
         tornado.web.Application.__init__(self, handlers)  # type: ignore
 
-    def log_request(self, handler: tornado.web.RequestHandler) -> None:  # skipcq: PTC-W0049
-        pass
+    def log_request(self, handler: tornado.web.RequestHandler) -> None:
+        """Overrides the default implementation since we have our own logging setup."""
 
 
-# WebhookHandler, process webhook calls
 # pylint: disable=abstract-method
-class WebhookHandler(tornado.web.RequestHandler):
-    SUPPORTED_METHODS = ["POST"]  # type: ignore
+class TelegramHandler(tornado.web.RequestHandler):
+    """Handler that processes incoming requests from Telegram"""
 
-    def __init__(
```

Comment:
why remove the init?

Reply:
tornade want's you to do any actual initialization in `initialize` …